File: app/src/main/java/pt/epcc/alunos/al220007/bloqs/MainActivity.java
package pt.epcc.alunos.al220007.bloqs;

import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import pt.epcc.alunos.al220007.bloqs.db.Database;
import pt.epcc.alunos.al220007.bloqs.list.BroadcastReceiver;
import pt.epcc.alunos.al220007.bloqs.models.User;
import pt.epcc.alunos.al220007.bloqs.models.UserBroadcastReceiver;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
private static final String TAG = "MainActivity";

private static final int RESET_DB_BTN_ID = R.id.reset_db_btn;
private static final int LAYOUT = R.layout.activity_main;

private final BroadcastReceiver<User> receiver = new UserBroadcastReceiver();
private Button resetDatabaseButton;

@Override
protected void onCreate(Bundle savedInstanceState) {
	super.onCreate(savedInstanceState);
	setContentView(LAYOUT);

	this.register();

	Intent intent = new Intent(BroadcastReceiver.VERIFY_DB_ACTION);
	this.sendBroadcast(intent);

	this.findViews();

	Log.i(TAG, "onCreate");
}

private void findViews() {
	this.resetDatabaseButton = this.findViewById(RESET_DB_BTN_ID);
	this.resetDatabaseButton.setOnClickListener(this);
}

private void register() {
	IntentFilter filter = new IntentFilter(BroadcastReceiver.VERIFY_DB_ACTION);
	filter.addCategory(Intent.CATEGORY_DEFAULT);

	this.registerReceiver(this.receiver, filter);
}

@Override
protected void onResume() {
	super.onResume();

	this.register();

	Log.i(TAG, "onResume");
}

@Override
protected void onStop() {
	super.onStop();

	this.unregisterReceiver(this.receiver);

	Log.i(TAG, "onStop");
}

@Override
protected void onDestroy() {
	super.onDestroy();

	this.unregisterReceiver(this.receiver);

	Log.i(TAG, "onDestroy");
}

@Override
public void onClick(View v) {
	if (v == this.resetDatabaseButton) {
		deleteDatabase(Database.INSTANCE.getName());
		this.finish();
	}
}
}

File: app/src/main/java/pt/epcc/alunos/al220007/bloqs/MainActivity.kt
package pt.epcc.alunos.al220007.bloqs

import android.content.Context
import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Spinner
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.ui.platform.ComposeView
import androidx.multidex.MultiDex
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import com.google.android.material.bottomnavigation.BottomNavigationView
import pt.epcc.alunos.al220007.bloqs.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    private lateinit var actionBar: ComposeView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
    }

    @OptIn(ExperimentalMaterial3Api::class)
    override fun onResume() {
        super.onResume()

        val navView: BottomNavigationView = binding.navView

        val navController = findNavController(R.id.nav_host_fragment_activity_main)

        val appBarConfiguration = AppBarConfiguration(
            setOf(
                R.id.navigation_bloqs
            )
        )

        setupActionBarWithNavController(navController, appBarConfiguration)
        navView.setupWithNavController(navController)

        val instance: Spinner = binding.instance
        val items = listOf("Item 1", "Item 2", "Item 3", "Item 4")
        val adapter = ArrayAdapter(applicationContext, R.layout.instance_item, items)
        instance.adapter = adapter
//        actionBar = binding.actionBar
//        actionBar.setContent {
//            Button(onClick = { }) {
//
//            }
//
//            TextField(
//                value = "https://bloqsenjin-ten.vercel.app/api/rest/bloq",
//                onValueChange = { v -> })
//
//            SearchBar(
//                query = "",
//                onQueryChange = { q -> },
//                onSearch = { q -> },
//                active = false,
//                onActiveChange = { active -> }) {
//
//            }
//        }
    }

    override fun attachBaseContext(newBase: Context?) {
        super.attachBaseContext(newBase)
        MultiDex.install(this)
    }
}